app/src/main/java/org/openobservatory/ooniprobe/data/TestData.java

```diff
@@ -179,6 +181,15 @@ else if (testName.compareTo(OONITests.WEB_CONNECTIVITY) == 0) {
                         w.set_options("save_real_probe_cc", boolToString(include_cc));
                         w.set_options("no_collector", boolToString(!upload_results));
                         w.set_options("collector_base_url", collector_address);
+                        w.on_progress(new org.openobservatory.measurement_kit.nettests.ProgressCallback() {
+                            @Override
+                            public void callback(double percent, String msg) {
+                                System.out.println("test progress "+ percent);
+                                currentTest.progress = (int)percent*100;
+                                //TODO crases here maybe we need to implement the LocalBroadcastManager?
+                                //TestData.getInstance(activity).notifyObservers();
```

Comment:
@lorenzoPrimi what kind of crash is this? A segfault or a Java exception? Is there a stacktrace?